teslajsonpy/controller.py

```diff
@@ -169,59 +166,46 @@ def valid_result(result):
         "%s: "
         "wake_up needed for %s -> %s "
         "Info: args:%s, kwargs:%s, "
-        "ID:%s, car_online:%s "
+        "ID:%s, state:%s "
         "is_wake_command:%s, is_wake_api:%s wake_if_asleep:%s",
         instance._id_to_vin(car_id)[-5:],
         wrapped.__name__,
         result,
         args,
         kwargs,
         car_id if car_id else None,
-        instance.car_online if instance.car_online else None,
+        instance.get_car_state(car_id=car_id) if car_id else None,
         is_wake_command,
         is_wake_api,
         kwargs.get("wake_if_asleep"),
     )
-    # instance.car_online[instance._id_to_vin(car_id)] = False
-    instance.set_car_online(car_id=car_id, online_status=False)
-    while (
-        kwargs.get("wake_if_asleep")
-        and
-        # Check online state
-        (
-            car_id is None
-            or (
-                not instance._id_to_vin(car_id)
-                or not instance.is_car_online(car_id=car_id)
-            )
-        )
+    while kwargs.get("wake_if_asleep") and (
+        car_id is None
+        or (not instance._id_to_vin(car_id) or instance.is_car_asleep(car_id=car_id))
     ):
         _LOGGER.debug("Attempting to wake up")
-        result = await instance._wake_up(car_id)
+        wake_success = await instance._wake_up(car_id)
         _LOGGER.debug(
             "%s: %s: Wake Attempt(%s): %s. Next attempt in %s",
             instance._id_to_vin(car_id)[-5:],
             wrapped.__name__,
             retries,
-            result,
+            wake_success,
             15 + sleep_delay ** (retries + 2),
```

Comment:
It may make sense to revisit the base wake time since we're editing this part. I believe the app is using something larger like 30. I think recent testing shows you can't ever get the car woken up by the first wake attempt.

Reply:
good idea, I also saw that the first attempt usually failed, I'll collect some real world data and adjust it.

Reply:
Oh man I'm glad I dug into this more, it's not that it takes long to wake, it's that the logic was using the response from the wake command before retrying...it should update to get a new response before retying to wake.

I wrote a script to test this and found that it takes less than 10 seconds to wake up, and only 1 command needs to be sent (unless it fails).

```
waking car...
state after wake attempt: asleep, time: 0.2
Updating...
state after refresh: asleep, time: 0.9
[...]
Updating...
state after refresh: asleep, time: 8.7
Updating...
state after refresh: online, time: 9.5
```

Doing this I also realized the entire wake_up decorator can be replaced with 3 lines. I guess a decorator was useful in the old version if it was used in several places, but it's only used in `Controller.api()` now, so it can be massively simplified. I'll have an update later today, I want to do lots of real-world testing with this change now that it's going to affect the wakeup logic

Reply:
ok I pushed a big cleanup of the wakeup logic.
It took me most of the day to understand what the old decorator was trying to do, and after realizing that it was quite neglected and optimized, I condensed it into Controller.api(), and also fixed a few minor edge cases such as if `wake_if_asleep` is true but the car has went to sleep after the latest `get_vehicles`, or `wake_if_asleep` is true for an api request that does not use/have the vehicle ID (it's not possible to wake something without it'd ID).

Real-world testing so far has been really promising, wake ups are faster, the data is updated faster, and less "WAKE_UP" api requests are made.

I also cleaned up about 100 lines of redundant parameters in car.py, and ensured that the command isn't attempted if the vehicle is asleep and we don't want to wake it, along with some better wake_if_asleep determination to not wake the car if we know it's a pointless command (such as turning the climate off. if the car is asleep, we know the climate is off)

Reply:
I guess that last change will help close https://github.com/alandtse/tesla/issues/379. Do you think your changes cover all such cases or just the few you noticed?

Reply:
Ahh, I missed that one in particular, I'll go through and add value comparisons to fix this for number entities as well